### ISSUE TITLE
Question / Feature: Use multiple call recorders at the same time

### ISSUE THREAD

**Issue by ElsAr4e** [chenxiaolong/BCR#524]
If you are using a rooted device is it possible that you use multiple call recorders at the same time to record one call?

For example:

- Basic Call Recorder + Samsung phone built in call recorder (Maybe after you've changed the SCS)
- Basic Call Recorder + ACR phone with ACR Phone Helper (APH) Magisk Module
- Basic Call Recorder + Screen recorder with root access that records internal audio

or other combinations like
Basic Call Recorder + Samsung phone built in call recorder (Maybe after you've changed the SCS) + Screen recorder with root access that records internal audio

----
If it's possible did you test such scenarios (different Android versions)?

If it's not possible will the OS and/or Basic Call Recorder display an error message (if the Basic Call Recorder tries to record a call that is already recorded by another app or if the Basic Call Recorder records a call and another app tries to record the same call as well)?

**Comment by chenxiaolong** [comment]
This will depend on the specific device you're using. You'll have to try it to see how your device will behave.

In general, I'd expect it to not work properly. In the devices I've owned since I initially wrote BCR, I've seen:
* it work fine (rarely)
* one recording has only one side of the audio and the other recording has no audio
* both recordings only have one side of the audio
* no audio in either recording
* the audio driver failing with an error in both recordings

> Basic Call Recorder + Screen recorder with root access that records internal audio

I'm not aware of any screen recorders that record the `VOICE_CALL` audio stream. It's only when two apps try to record `VOICE_CALL` at the same time that you'll run into issues.

**Comment by ElsAr4e** [comment]
Is it technically possible that an app checks (rooted vs. non-rooted) if another app already records one call?

If yes - for example can / should Basic Call Recorder display a warning message that another app already records the call and ask the user if the user really wants that Basic Call Recorder records the call too?

**Comment by chenxiaolong** [comment]
No, as far as I'm aware, there's no way to detect if another app is already recording `VOICE_CALL`.

**Comment by ElsAr4e** [comment]
Can you use the Basic Call Recorder (on a rooted device) together with a screen recorded (for rooted / non-rooted devices) that captures the internal sound?

For example if you want to record a phone call and at the same time a meeting of Jitsi-Meet (or BraveTalk) or YouTube video using a screen recorder this should be possible?

---

And what if you use ACR Phone app and the rooted option ACR Phone Helper (APH) Magisk Module to record a voip call like WhatsApp?

Can you use Basic Call Recorder at the same time to record a GSM call?

**Comment by chenxiaolong** [comment]
Only regular (cellular) phone calls use `VOICE_CALL`. All other audio (VOIP, audio playback, etc.) use different audio streams. Recording those with a different app while BCR is recording a regular phone call should be fine.

**Comment by ElsAr4e** [comment]
Even if ACR-Phone app requires Android Connection Service API to be used by WhatsApp etc. (VOICE_CALL is NOT needed)? https://nllapps.com/apps/cb/voip-call-recording.htm

Capturing a WhatsApp call using ACR-Phone and a regular cellular phone call at the same time using BCR should be no problem (no errors)?

--
And if you use the Boldbeast Android VoIP Call Recorder https://www.boldbeast.com/android/voip_call_recorder.html (other technology) still no issue if you record a regular phone call and a voip app like WhatsApp at the same time?

**Comment by chenxiaolong** [comment]
Regular calls use the `VOICE_CALL` stream and VOIP apps use either the `VOICE_COMMUNICATION` or `MIC` streams. Two recordings at the same time should be fine as long as they're recording two different streams.

(Android's connection service APIs aren't relevant to this question. The API is only used for detecting when a call starts and stops. Recording happens via the AudioRecord APIs.)

**Comment by ElsAr4e** [comment]
Thanks for all these answers.

So the only thing you can do is to display a message the first time BCR is installed like "Keep in mind BCR should not be used with other regular cellular phone call recorders to record the same call at the same time"? (Would such a notification make sense?)

**Comment by chenxiaolong** [comment]
No problem!

I'd prefer not to add another notification, but I do think it's good to mention. I'll make a note of it in the installation instructions.

**Comment by ElsAr4e** [comment]
@chenxiaolong FYI: I've done hands-on research and tried things out purely for experimental purposes.

I used a Samsung Galaxy A52s 5G (A528B/DS) running One UI 6.1, Magisk 30.6, and BCR 2.0:
I was able to record regular phone calls flawlessly and without any technical issues using either the Samsung dialer’s built-in call recording with BCR, or ACR Phone with the APH Magisk root module in combination with BCR.

However, with BCR I used WAV (PCM) at 48,000 Hz (user-defined) in either mono or stereo, whereas the other dialers always recorded in .m4a format—even when I set the recording quality to maximum in ACR Phone, it had no effect.
